universal-login-wallet/src/ui/react/CreateAccount/CreateFlow.tsx

```diff
@@ -0,0 +1,76 @@
+import React, {useContext, useState} from 'react';
+import {Redirect} from 'react-router';
+import {useMutableState} from 'react-use-mutable';
+import {useServices} from '../../hooks';
+import {CreateAccount} from './CreateAccount';
+import {WalletModalContext} from '../../../core/entities/WalletModalContext';
+import {ETHER_NATIVE_TOKEN, GasParameters, INITIAL_GAS_PARAMETERS} from '@universal-login/commons';
+import {hideTopUpModal} from '../../../core/utils/hideTopUpModal';
+import Modal from '../Modals/Modal';
+import {ModalWrapper, TopUp, useProperty, WaitingForDeployment} from '@universal-login/react';
+import {ModalTxnSuccess} from '../Modals/ModalTxnSuccess';
+
+export function CreateFlow() {
+  const modalService = useContext(WalletModalContext);
+  const {sdk, walletService} = useServices();
+  const [getGasParameters, setGasParameters] = useMutableState<GasParameters>(INITIAL_GAS_PARAMETERS);
```

Comment:
Interesting case here. Because of the way javascript functions capture variables, using normal `useState` here would result in `gasPrice` not being updated midway through `onCreateClick` execution. This is because with each render `useState` technically returns a new variable, and the old ones keep their old values. `useMutableState` on the other hand returns a getter. This makes it possible to make sure that call to this getter always returns the latest values that was set.

Here's the implementation of `useMutableState` for reference:
https://github.com/Marik-D/react-use-mutable/blob/master/src/useMutableState.ts

Reply:
👍

Reply:
Can we use bind instead?
```ts
return <CreateAccount onCreateClick={onCreateClick.bind(this)}/>;
```

Reply:
How we can solve in a simpler way?

Reply:
```ts
const deploymentService = new DeploymentService(walletService);
...
<... onGasPrametersChanged=deploymentService.setGasParameters() ...>
...
<... onCreateCreateClick={deploymentService.deploy} ...>
```